src/saltext/vmware/modules/esxi.py

```diff
@@ -845,6 +845,326 @@ def get_ntp_config(
+
+    datacenter_name
+        Filter by this datacenter name (required when cluster is specified)
+
+    cluster_name
+        Filter by this cluster name (optional)
+
+    host_name
+        Filter by this ESXi hostname (optional)
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    .. code-block:: bash
+
+        salt '*' vmware_esxi.remove_user user_name=foo
+    """
+    log.debug("Running vmware_esxi.remove_user")
+    ret = {}
+    if not service_instance:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+    hosts = utils_esxi.get_hosts(
+        service_instance=service_instance,
+        host_names=[host_name] if host_name else None,
+        cluster_name=cluster_name,
+        datacenter_name=datacenter_name,
+        get_all_hosts=host_name is None,
+    )
+
+    try:
+        for h in hosts:
+            h.configManager.accountManager.RemoveUser(user_name)
+            ret[h.name] = True
+        return ret
+    except (
+        vim.fault.InvalidState,
+        vim.fault.NotFound,
+        vim.fault.HostConfigFault,
+        vmodl.fault.InvalidArgument,
+        salt.exceptions.VMwareApiError,
+        vim.fault.UserNotFound,
+    ) as exc:
+        raise salt.exceptions.SaltException(str(exc))
+
+
+def get_user(
+    user_name=None,
+    datacenter_name=None,
+    cluster_name=None,
+    host_name=None,
+    service_instance=None,
+):
+    """
+    Get local user on matching ESXi hosts.
+
+    user_name
+        Filter by this user name (required).
+
+    datacenter_name
+        Filter by this datacenter name (required when cluster is specified)
+
+    cluster_name
+        Filter by this cluster name (optional)
+
+    host_name
+        Filter by this ESXi hostname (optional)
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    .. code-block:: bash
+
+        salt '*' vmware_esxi.get_user user_name=foo
+    """
+    log.debug("Running vmware_esxi.get_user")
+    ret = {}
+    if not service_instance:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+    hosts = utils_esxi.get_hosts(
+        service_instance=service_instance,
+        host_names=[host_name] if host_name else None,
+        cluster_name=cluster_name,
+        datacenter_name=datacenter_name,
+        get_all_hosts=host_name is None,
+    )
+
+    try:
+        for h in hosts:
+            users = h.configManager.userDirectory.RetrieveUserGroups(
+                searchStr=user_name,
+                belongsToGroup=None,
+                belongsToUser=None,
+                domain=None,
+                exactMatch=True,
+                findUsers=True,
+                findGroups=True,
+            )
+            for user in users:
+                ret[h.name] = {
+                    user.principal: {
```

Comment:
I'm assuming principal is unique, maybe the `username` - is that what it is? It's an odd term.

Reply:
yes, that is the user name. :)

Reply:
It might be more clear to use `user_name` instead of `user.principal`, then

Reply:
The value of user.principal which contains the user_name is the key. I am not following your comment. we are not using `user.principal` literal in the dict.

Reply:
It's not necessary, but if `user_name == user.principal`, then it might be more clear to use `user_name` because it's less obvious what `user.principal` is.

Reply:
oh like that ok : 👍

Reply:
but it conflicts for others like `user_name.group`, `user_name.id` as opposed to `user.*`. Also user is an object, so user_name wouldn't be a correct usage. Let me know what you think.

Reply:
will add comment